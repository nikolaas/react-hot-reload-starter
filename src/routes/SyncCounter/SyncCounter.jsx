import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { BaseLayout } from '../../components/BaseLayout';
import { Article } from '../../components/Article';
import { increment } from './SyncCounter.actions';

export const RawSyncCounter = ({ counter, onIncrement }) => {
    return (
        <BaseLayout>
            <Article title="Counter">
                <p className="SyncCounter__value">Counter: <span style={{ color: '#f00' }}>{counter}</span></p>
                <button onClick={() => onIncrement()}>Increment</button>
            </Article>
        </BaseLayout>
    );
};

RawSyncCounter.propTypes = {
    counter: PropTypes.number,
    onIncrement: PropTypes.func.isRequired,
};

RawSyncCounter.defaultProps = {
    counter: 0,
};

const mapStateToProps = state => {
    const { syncCounter } = state;
    return { counter: syncCounter };
};

const mapDispatchToProps = {
    onIncrement: increment,
};

export default connect(mapStateToProps, mapDispatchToProps)(RawSyncCounter);
