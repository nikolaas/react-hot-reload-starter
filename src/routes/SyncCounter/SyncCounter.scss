@import '../../theme/fonts';

.SyncCounter {
  font-size: $font-size;
}
