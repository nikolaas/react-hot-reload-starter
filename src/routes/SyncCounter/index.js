import { registerAsyncReducer } from '../../store';
import SyncCounter from './SyncCounter';
import reducer from './SyncCounter.reducer';

registerAsyncReducer('syncCounter', reducer);

export default SyncCounter;
