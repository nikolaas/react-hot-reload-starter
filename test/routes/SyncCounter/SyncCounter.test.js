import React from 'react';
import { shallow } from 'enzyme';
import { RawSyncCounter } from '../../../src/routes/SyncCounter/SyncCounter';

describe('Sync counter tests', () => {
    it('should render default value when value is not specified', () => {
        const component = shallow(<RawSyncCounter onIncrement={jest.fn()}/>);
        expect(component.find('.SyncCounter__value').text()).toMatchSnapshot();
    });

    it('should render specified value when value is specified', () => {
        const component = shallow(<RawSyncCounter onIncrement={jest.fn()} counter={10}/>);
        expect(component.find('.SyncCounter__value').text()).toMatchSnapshot();
    });

    it('should trigger onIncrementEvent when increment button was clicked', () => {
        const incrementSpy = jest.fn();
        const wrapper = shallow(<RawSyncCounter onIncrement={incrementSpy}/>);
        wrapper.find('button').simulate('click');
        expect(incrementSpy).toHaveBeenCalled();
    });
});
