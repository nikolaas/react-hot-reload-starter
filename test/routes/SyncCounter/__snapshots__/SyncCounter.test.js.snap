// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sync counter tests should render default value when value is not specified 1`] = `"Counter: 0"`;

exports[`Sync counter tests should render specified value when value is specified 1`] = `"Counter: 10"`;
